packages/tinacms/src/toolkit/fields/components/reference/reference-select.tsx

```diff
@@ -182,33 +183,32 @@ const ComboboxDemo: React.FC<ReferenceSelectProps> = ({
                     heading={collection}
                   >
                     <CommandList>
-                      {edges.map(
-                        ({
-                          node: {
-                            id,
-                            _internalSys: { filename },
-                            _values: { title, name },
-                          },
-                        }) => (
+                      {edges.map(({ node }) => {
+                        const { id, _values } = node
+
+                        return (
                           <CommandItem
                             key={`${id}-option`}
-                            value={id || title}
+                            value={id}
                             onSelect={() => {
                               setValue(id)
                               setOpen(false)
                             }}
                           >
                             <div className="flex flex-col">
-                              <span className="font-semibold text-sm">
-                                {title || name || id}
-                              </span>
-                              {(title || name) && (
-                                <span className="text-x">{filename}</span>
-                              )}
+                              <div>
+                                {field?.ui?.selectComponents && _values ? (
+                                  field.ui.selectComponents(_values, {
```

Comment:
As we discussed, it might be nice to give the callback value a `values` object but also all of the `_sys` values we give you in the GraphQL response.

```ts
{
  values: Record<srtring, unknown>,
  _sys: {
   filename: string
   title: string // this would be used if `isTitle` is set on the collection
   breadcrumbs: string[],
   extension: string,
   // ... etc
 }
}

Reply:
As per our conversation, we will tackle the custom search in a different pr (current pr is for custom reference e component and the search still work based on filename )

PBI - https://github.com/tinacms/tinacms/issues/4738